boa/src/builtins/object/mod.rs

```diff
@@ -98,7 +98,11 @@ impl Object {
     /// Define a property in an object
     pub fn define_property(_: &Value, args: &[Value], ctx: &mut Context) -> Result<Value> {
         let obj = args.get(0).expect("Cannot get object");
-        let prop = args.get(1).expect("Cannot get object").to_string(ctx)?;
+        let prop = args
+            .get(1)
+            .expect("Cannot get object")
+            .to_property_key(ctx)
+            .expect("Cannot be used as property key");
```

Comment:
```suggestion
            .to_property_key(ctx)?;
```
We should use the `?` try operator to propagate the error if one is thrown, instead of panicking.